GameFighter/Battle.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace GameFighter
+{
+    public static class Battle
+    {
+        public static bool Fight(Warrior warrior1, Warrior warrior2)
+        {
+            if (warrior1 == null || warrior2 == null)
+            {
+                throw new NotImplementedException("warrior1 and warrior2 must be initialised!");
```

Comment:
Why do you use NotImplementedException? It's more obvious to use ArgumentNullException. Or you can combine both if statements and use InvalidArgumentException